source/SIL.AppBuilder.Portal.Frontend/src/ui/components/product-artifact/artifacts.tsx

```diff
@@ -1,111 +1,70 @@
-import { withData as withOrbit } from 'react-orbitjs';
-import ProductIcon from '@ui/components/product-icon';
-
-import {
-  ProductDefinitionResource,
-  ProductArtifactResource,
-  ProductResource,
-  ProductBuildResource,
-  attributesFor,
-} from '@data';
-
+import { useOrbit } from 'react-orbitjs';
 import EmptyLabel from '@ui/components/labels/empty';
 import { isEmpty } from '@lib/collection';
-import { withTranslations, i18nProps } from '@lib/i18n';
+import { useTranslations } from '@lib/i18n';
 
 import Artifact from './artifact';
 import Header from './header';
 
-interface IExpectedProps {
-  product: ProductResource;
-  productBuild: ProductBuildResource;
-}
-
-interface IOwnProps {
-  productDefinition: ProductDefinitionResource;
-  artifacts: ProductArtifactResource[];
-}
-
-interface IState {
-  areArtifactsVisible: boolean;
-}
-
-type IProps = IOwnProps & i18nProps;
-
-class ProductArtifact extends React.Component<IProps, IState> {
-  state = { areArtifactsVisible: false };
+import { useToggle } from '~/lib/hooks';
 
-  toggleShowArtifacts = () => {
-    this.setState({ areArtifactsVisible: !this.state.areArtifactsVisible });
-  };
+export default function ProductArtifact({ product, productBuild }) {
+  const { t } = useTranslations();
+  const { dataStore } = useOrbit();
+  const [areArtifactsVisible, toggleShowArtifacts] = useToggle(false);
 
-  render() {
-    const { artifacts, t } = this.props;
-    const { areArtifactsVisible } = this.state;
+  const productDefinition = dataStore.cache.query((q) =>
+    q.findRelatedRecord(product, 'productDefinition')
+  );
+  let artifacts = [];
 
-    return (
-      <div data-test-product-artifacts className='product-artifact w-100 m-b-lg'>
-        <Header
-          {...this.props}
-          onClick={this.toggleShowArtifacts}
-          isCollapsed={areArtifactsVisible}
-        />
+  if (productBuild) {
+    artifacts = dataStore.cache.query((q) =>
+      q.findRelatedRecords(productBuild, 'productArtifacts')
+    );
+  }
 
-        {areArtifactsVisible && (
-          <div data-test-artifact-list-container>
-            <EmptyLabel
-              className='m-t-lg m-l-lg m-b-lg'
-              condition={!isEmpty(artifacts)}
-              label={t('project.products.noArtifacts')}
-            >
-              <div className='flex p-l-md p-t-sm p-b-sm p-r-md gray-text bold'>
-                <div className='flex-70'>
-                  <span>{t('project.products.filename')}</span>
+  return (
+    <div data-test-product-artifacts className='product-artifact w-100 m-b-lg'>
+      <Header
+        {...{
+          productDefinition,
+          artifacts,
+          product,
+        }}
+        onClick={toggleShowArtifacts}
+        isCollapsed={areArtifactsVisible}
+      />
+
+      {areArtifactsVisible && (
+        <div data-test-artifact-list-container>
+          <EmptyLabel
+            className='m-t-lg m-l-lg m-b-lg'
```

Comment:
personal preference and something I don't expect to see changed but I really dislike these classNames.  I am sure it makes sense once you understand the convention but all I get from it is something is happening at the large breakpoint

Reply:
Yeah, this isn't my first preference for styling _everything_, but it's what we have without a design/ui language.

Fwiw, the pattern is from bootstrap: https://getbootstrap.com/docs/4.3/utilities/spacing/